include/cutlass/epilogue/collective/builders/xe_builder.inl

```diff
@@ -166,14 +174,21 @@ template <
       #endif
       static_assert(is_static<TileShape_MNK>::value);
       static_assert(cute::is_any_of_v<ElementC, float, bfloat16_t, void>, "ElementC needs to be float or bfloat for the Intel pipeline");
-      
-      using EpilogueSchedule = std::conditional_t<cute::is_same_v<EpilogueScheduleType, EpilogueScheduleAuto>, 
+      using EpilogueSchedule = std::conditional_t<cute::is_same_v<EpilogueScheduleType, EpilogueScheduleAuto>,
                                                   IntelXeXMX16,
                                                   EpilogueScheduleType>;
       static constexpr bool IsGroup = cute::is_same_v<EpilogueSchedule, IntelXeXMX16Group>;
-      using DispatchPolicy = std::conditional_t<IsGroup, 
+      using DispatchPolicy = std::conditional_t<IsGroup,
                                                 IntelXeXMX16Group,
                                                 IntelXeXMX16>;
+      using LayoutC = std::remove_pointer_t<GmemLayoutTagC>;
+      using LayoutD = std::remove_pointer_t<GmemLayoutTagD>;
+      static constexpr bool row_majorish_C = detail::is_rowmajorish<LayoutC>::value;
+      static constexpr bool row_majorish_D = detail::is_rowmajorish<LayoutD>::value;
```

Comment:
I think the logic here could be a lot simpler if you first grabbed `StrideC` and `StrideD` tuples using the `std::conditional_t` below, then asserted they were rowmajorish.

Reply:
ye, I think you're right. I've reworked it a bit, what do you think?